### DISCUSSION TITLE
Asset packs forced to be re-downloaded when AAB updated

### DISCUSSION THREAD

**Discussion by a-l-e-x-d-s-9** [jelte/be.khepri.play.assetdelivery.addressables#21] [General]
I'm not sure if I'm missing something, but when I update the AAB in Google Store, my device seems to download the base and all the assets packs again. So even if the assets didn't change at all, they re-downloaded anyway. 
[From this post, it looks like intended behavior.](https://github.com/google/play-unity-plugins/issues/19#issuecomment-632451516)
But in this case, the whole Google Play Asset Delivery system is kinda useless when updating AAB.
Am I wrong?

**Comment by a-l-e-x-d-s-9** [comment]
@jelte Can you please comment on this topic?

**Reply by jelte** [reply]
Hey, 
I'm not sure as this is internal google play logic. But I assume they use the same kind of functionality that they do with their normal AABs.

Which is that they do some kind of diff between the two when you download them and only update the content that is changed internally. You can see this behaviour when you push a new AAB (without assetpacks) for the same content and you download it. 
It will still say the total size. but while downloading it only downloads a fraction of that.
I assume that as long as your asset packs don't change (name and content) then it should be using that logic.

Again, this is internal google play logic, so I can't really know for sure.

**Comment by a-l-e-x-d-s-9** [comment]
I just wanted to ask if you have noticed this behavior in your experience too, maybe you haven't noticed.
I too assumed that Google Play Store isn't downloading unnecessary asset bundles, if the bundles didn't change between AABs updates.
But from what I see when updating, all on-demand bundles are automatically downloaded by Google Play Store, even if they didn't change at all. 
While updating, Google Play store showing the percentage of the download out of large AAB size that definitely includes all bundle sizes. I don't think that it just report the full size, but don't download it all, while reporting that it does. 
Also, if you look at [this post](https://github.com/google/play-unity-plugins/issues/19#issuecomment-632451516), it sounds to me that it's exactly what Google Play logic is doing.
